admin/services/marking.service.js

```diff
@@ -43,23 +44,35 @@ markingService.mark = async function (completedCheck) {
     processedAt: moment.utc()
   }
 
+  const checkForm = await checkFormDataService.sqlFindOneById(completedCheck.checkForm_id)
+
+  if (!checkForm || !checkForm.formData) {
+    throw new Error('check form data missing or not found')
+  }
+
+  const formData = JSON.parse(checkForm.formData)
+
   // Store the mark for each answer
   const answers = []
   let questionNumber = 1
 
-  for (let answer of completedCheck.data.answers) {
+  for (let question of formData) {
+    const currentIndex = questionNumber - 1
+    const answer = completedCheck.data.answers[currentIndex]
+    const currentAnswer = answer && answer.answer
     const data = R.clone(answer)
-    data.answer = R.slice(0, 60, answer.answer)
+    // Immediately break when answer is not found based on question index
+    if (!data) break
```

Comment:
Why do you want to use `break` here? 

We still need to mark the question.  If we can't find the question then they get 0 marks, which we should store.

Reply:
@jon-shipley Valid point, it is updated now